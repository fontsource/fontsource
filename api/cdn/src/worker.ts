import { error } from 'itty-router';

import router, { corsify } from './router';

export default {
	async fetch(
		request: Request,
		env: Env,
		ctx: ExecutionContext,
	): Promise<Response> {
		return await router
			.handle(request, env, ctx)
			.catch((error_) => {
				console.error(error_);
				const resp = error(error_);
				// Our jsDelivr proxy will fallback to one hour if no cache-control is set
				resp.headers.set('Cache-Control', 'max-age=0');
				return resp;
			})
			.then(corsify);
	},
};
