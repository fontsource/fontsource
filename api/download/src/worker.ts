import router from './router';
import { error } from 'itty-router';

export default {
	async fetch(
		request: Request,
		env: Env,
		ctx: ExecutionContext
	): Promise<Response> {
		return router.handle(request, env, ctx).catch((err) => {
			console.error(err);
			return error(err);
		});
	},
};
