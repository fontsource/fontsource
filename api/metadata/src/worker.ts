import fontsRouter from './fonts/router';
import fontlistRouter from './fontlist/router';
import downloadRouter from './download/router';
import { error } from 'itty-router';

export default {
	async fetch(
		request: Request,
		env: Env,
		ctx: ExecutionContext
	): Promise<Response> {
		try {
			const url = new URL(request.url);

			if (url.pathname.startsWith('/fontlist')) {
				return fontlistRouter.handle(request, env, ctx).catch((err) => {
					console.error(err);
					return error(err);
				});
			}

			if (url.pathname.startsWith('/v1/fonts')) {
				return fontsRouter.handle(request, env, ctx).catch((err) => {
					console.error(err);
					return error(err);
				});
			}

			if (url.pathname.startsWith('/v1/download')) {
				return downloadRouter.handle(request, env, ctx).catch((err) => {
					console.error(err);
					return error(err);
				});
			}

			return error(
				404,
				'Not Found. Please refer to the Fontsource API documentation: https://fontsource.org/docs/api'
			);
		} catch (e) {
			console.error(e);
			return error(500, 'Internal Server Error.');
		}
	},
};
