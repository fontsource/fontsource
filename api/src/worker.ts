import fontsRouter from './fonts/router';

export default {
	async fetch(
		request: Request,
		env: Env,
		ctx: ExecutionContext
	): Promise<Response> {
		const url = new URL(request.url);

		if (url.pathname.startsWith('/v1/fonts/')) {
			return fontsRouter.handle(request);
		}

		return new Response('Not Found.', { status: 404 });
	},
};
