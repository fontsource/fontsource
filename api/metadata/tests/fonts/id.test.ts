import { beforeEach, expect, it } from 'vitest';
const describe = setupMiniflareIsolatedStorage();
import worker from '../../src/worker';
import { mockMetadata } from '../helpers';

describe('fonts id worker', () => {
	const env = getMiniflareBindings() as Env;
	const ctx = new ExecutionContext();

	beforeEach(async () => {
		mockMetadata();
	});

	it('should return normal id response', async () => {
		const request = new Request('http://localhost:8787/v1/fonts/akshar');
		const response = await worker.fetch(request, env, ctx);
		expect(response.status).toBe(200);
		const data = await response.json();
		expect(data).toMatchSnapshot();
	});

	it('should return 404 for invalid id', async () => {
		const request = new Request('http://localhost:8787/v1/fonts/invalid-id');
		const response = await worker.fetch(request, env, ctx);
		expect(response.status).toBe(404);
	});
});
