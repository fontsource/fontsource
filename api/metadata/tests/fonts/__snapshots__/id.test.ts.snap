// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fonts id worker > should return normal id response 1`] = `
{
  "category": "sans-serif",
  "defSubset": "latin",
  "family": "Akshar",
  "id": "akshar",
  "lastModified": "2023-03-21",
  "license": "OFL-1.1",
  "npmVersion": "5.0.0",
  "source": "https://github.com/google/fonts",
  "styles": [
    "normal",
  ],
  "subsets": [
    "devanagari",
    "latin",
    "latin-ext",
  ],
  "type": "google",
  "unicodeRange": {
    "devanagari": "U+0900-097F,U+1CD0-1CF9,U+200C-200D,U+20A8,U+20B9,U+25CC,U+A830-A839,U+A8E0-A8FF",
    "latin": "U+0000-00FF,U+0131,U+0152-0153,U+02BB-02BC,U+02C6,U+02DA,U+02DC,U+0300-0301,U+0303-0304,U+0308-0309,U+0323,U+0329,U+2000-206F,U+2074,U+20AC,U+2122,U+2191,U+2193,U+2212,U+2215,U+FEFF,U+FFFD",
    "latin-ext": "U+0100-02AF,U+0300-0301,U+0303-0304,U+0308-0309,U+0323,U+0329,U+1E00-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF",
  },
  "variable": true,
  "variants": {
    "300": {
      "normal": {
        "devanagari": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-300-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-300-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-300-normal.woff2",
          },
        },
        "latin": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-300-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-300-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-300-normal.woff2",
          },
        },
        "latin-ext": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-300-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-300-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-300-normal.woff2",
          },
        },
      },
    },
    "400": {
      "normal": {
        "devanagari": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-400-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-400-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-400-normal.woff2",
          },
        },
        "latin": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-400-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-400-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-400-normal.woff2",
          },
        },
        "latin-ext": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-400-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-400-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-400-normal.woff2",
          },
        },
      },
    },
    "500": {
      "normal": {
        "devanagari": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-500-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-500-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-500-normal.woff2",
          },
        },
        "latin": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-500-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-500-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-500-normal.woff2",
          },
        },
        "latin-ext": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-500-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-500-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-500-normal.woff2",
          },
        },
      },
    },
    "600": {
      "normal": {
        "devanagari": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-600-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-600-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-600-normal.woff2",
          },
        },
        "latin": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-600-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-600-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-600-normal.woff2",
          },
        },
        "latin-ext": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-600-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-600-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-600-normal.woff2",
          },
        },
      },
    },
    "700": {
      "normal": {
        "devanagari": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-700-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-700-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/devanagari-700-normal.woff2",
          },
        },
        "latin": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-700-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-700-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-700-normal.woff2",
          },
        },
        "latin-ext": {
          "url": {
            "ttf": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-700-normal.ttf",
            "woff": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-700-normal.woff",
            "woff2": "https://cdn.jsdelivr.net/fontsource/fonts/akshar@latest/latin-ext-700-normal.woff2",
          },
        },
      },
    },
  },
  "version": "v9",
  "weights": [
    300,
    400,
    500,
    600,
    700,
  ],
}
`;
